### PR TITLE
Add dev.edfloreshz.CosmicTweaks

### PR DESCRIPTION
**Pull request by edfloreshz**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
 A tweaking tool for the COSMIC desktop. 

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by bbhtt** [comment]
You can start the build when the exception is up

**Comment by bbhtt** [comment]
bot, build dev.edfloreshz.CosmicTweaks

File: dev.edfloreshz.CosmicTweaks.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  "sdk": "org.freedesktop.Sdk",
+  "sdk-extensions": ["org.freedesktop.Sdk.Extension.rust-stable"],
+  "command": "cosmic-ext-tweaks",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--device=dri",
+    "--talk-name=com.system76.CosmicSettingsDaemon",
+    "--filesystem=xdg-config/cosmic:rw"
```
Comment (bbhtt) [inline dev.edfloreshz.CosmicTweaks.json:14]: You made the exception for ro but this is rw Which one is needed ?

Comment (edfloreshz) [reply dev.edfloreshz.CosmicTweaks.json:14]: Oops, sorry I missed that, rw is the one needed.

Comment (edfloreshz) [reply dev.edfloreshz.CosmicTweaks.json:14]: Here's the [PR](https://github.com/flathub-infra/flatpak-builder-lint/pull/487)

**Comment by bbhtt** [comment]
bot, build dev.edfloreshz.CosmicTweaks

**Comment by bbhtt** [comment]
Remove cosmic from categories, it was added last month to xdg-specs and no release has been made since then, so it is failing validation.

Also according to @jackpot51, they are going to use a different tag for marking apps suited for COSMIC see the discussion https://github.com/mariinkys/starrydex/pull/3#issuecomment-2380880405

**Comment by bbhtt** [comment]
You are also missing releases https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#release

**Comment by bbhtt** [comment]
I think for OnlyShowIn to work you need a new desktop-file-utils release as well.

**Comment by bbhtt** [comment]
I will update desktop-file-utils to the latest master commit. So you can just fix the metainfo file - remove cosmic from category and add releases.

**Comment by edfloreshz** [comment]
Thanks! Latest commit should fix the issues, let me know if there's anything else missing.

**Comment by bbhtt** [comment]
bot, build dev.edfloreshz.CosmicTweaks

**Comment by bbhtt** [comment]
```
install: cannot stat './res/app_icon.svg': No such file or directory

```

Looks like the icon is no more.

**Comment by edfloreshz** [comment]
Sorry about that, should be fixed now.

**Comment by bbhtt** [comment]
bot, build dev.edfloreshz.CosmicTweaks

**Comment by jackpot51** [comment]
@edfloreshz COSMIC category needs to be removed.

**Comment by edfloreshz** [comment]
Should be fixed now

**Comment by ahoneybun** [comment]
bot, build dev.edfloreshz.CosmicTweaks

**Comment by bbhtt** [comment]
Exit buttons seem to be invisible and only a shadow shows up on hover. Is that intended ?


https://github.com/user-attachments/assets/388ba0ce-0c7e-48fb-9584-aab560961f91

**Comment by bbhtt** [comment]
I also see a 

```
error sending request for url (https://cosmic-themes.org/api/themes/?order=name&limit=15&offset=15)
```

in terminal

**Comment by wiiznokes** [comment]
> @edfloreshz COSMIC category needs to be removed.

Did you mean in category or onlyShowIn ? Because onlyShowIn is somehow useful

**Comment by wiiznokes** [comment]
Or maybe we can use X-COSMIC instead. I haven't tested it tho

**Comment by bbhtt** [comment]
OnlyShowIn=COSMIC will work now

I bumped desktop-file-utils to the latest commit in infra.

**Comment by wiiznokes** [comment]
> Exit buttons seem to be invisible and only a shadow shows up on hover. Is that intended ?
> 
> 
> https://github.com/user-attachments/assets/388ba0ce-0c7e-48fb-9584-aab560961f91
> 
> 
> 
> 

https://github.com/pop-os/libcosmic/issues/550

**Comment by bbhtt** [comment]
A baseapp with just the icon theme right now is not worth it. If it grows to multiple shared dependencies that all libcosmic apps need then it's fine.

If bundling the icon theme in the app, solves it then it's fine to do that for now.

**Comment by wiiznokes** [comment]
Technically, with the onlyShowIn, I think it will works (at least it showed the icon when I was building it manually)

**Comment by bbhtt** [comment]
I guess that's true for this one, but other apps will need to do it that aren't specific to only cosmic DE

**Comment by wiiznokes** [comment]
Could it be an extension of `org.freedesktop.Sdk` ?

**Comment by bbhtt** [comment]
There is an extension point for that `org.freedesktop.Platform.Icontheme`, but it doesn't autodownload. What you want is the icon theme gets automatically downloaded for everyone but that's not possible.

If you want you can submit a `org.freedesktop.Platform.Icontheme.COSMIC` but people have to manually install it.

**Comment by edfloreshz** [comment]
Should I bring `OnlyShowIn=COSMIC` back? It was failing the last time it compiled.

**Comment by bbhtt** [comment]
Yea should work now.

**Comment by wiiznokes** [comment]
> There is an extension point for that `org.freedesktop.Platform.Icontheme`, but it doesn't autodownload. What you want is the icon theme gets automatically downloaded for everyone but that's not possible.
> 
> If you want you can submit a `org.freedesktop.Platform.Icontheme.COSMIC` but people have to manually install it.

So there is nothing we can specify in the manifest that will include the icon theme ?

**Comment by bbhtt** [comment]
Bundle it ?

**Comment by wiiznokes** [comment]
Yeah but i don't think the theme is cheap in size

**Comment by edfloreshz** [comment]
I'm noticing some issues with the app, themes are not being applied, I'll try to figure out what's happening before we merge this.

**Comment by bbhtt** [comment]
> Yeah but i don't think the theme is cheap in size

I think having a fallback icon here would solve this, unless you want to preserve same look and feel everywhere. 

If you are on COSMIC then you already have the theme installed with your DE and Flatpak will just use those from the host.

For people not on COSMIC either it uses a bundled version or a fallback icon theme.

If you want COSMIC icon theme for people not on COSMIC as well, then right now you have to:

1. define a new extension point in the GNOME runtime and Freedesktop Runtime `com.s76.COSMIC.IconTheme.FOobar` one that has `autodownload: true`. 
2. Then submit the theme as
`com.s76.COSMIC.IconTheme.FOobar.CosmicIconTheme`.

3. Then app developers can request it with:

```yaml
add-extensions:
 com.s76.COSMIC.IconTheme.FOobar.CosmicIconTheme
```

and it will get autoinstalled whenever people install it.

A baseapp will not help here as stuff from the baseapp is copied to the target app as is. So there won't be in savings in size.

**Comment by bbhtt** [comment]
> I'm noticing some issues with the app, themes are not being applied, I'll try to figure out what's happening before we merge this.

Ok let men know when it's ready again.

**Comment by wiiznokes** [comment]
Ok, seems like a nice solution for the future

**Comment by wiiznokes** [comment]
> I'm noticing some issues with the app, themes are not being applied, I'll try to figure out what's happening before we merge this.

Good luck with that, i already tried to debugging this without success

**Comment by bbhtt** [comment]
As a note, if you are adding new extension points to the fdsdk runtime, they can only be part of the next major release.

ie. If you add it now, it will be available for 25.08 + (October 2025).

These are part of the runtime ABI/API and fdsdk maintains both forward and backwards ABI.

**Comment by edfloreshz** [comment]
> error sending request for url (https://cosmic-themes.org/api/themes/?order=name&limit=15&offset=15)

This is due to it not having network access, I know `--share=network` is not allowed in the manifest, how should I handle this? @bbhtt

**Comment by bbhtt** [comment]
It's only not allowed as a `build-arg` ie. you can't request network at build time. At runtime in `finish-args` it's completely fine if you need it.

**Comment by edfloreshz** [comment]
Okay, thanks, that should do it, we're ready for building and if everything goes well, publishing.

**Comment by bbhtt** [comment]
bot, build dev.edfloreshz.CosmicTweaks

**Comment by edfloreshz** [comment]
Working as expected, okay to merge

**Comment by bbhtt** [comment]
> Yeah but i don't think the theme is cheap in size

https://github.com/flathub/flathub/pull/5648 bundled the theme and it's 9.5 MB in total. If the entirety of the icon theme is only 4-5 MB it's better to just bundle it than going through the overhead of create an extension point, then submit the theme as extension.

**Pull request closed without merging**